pkg/server/commands/connectedobjects/connected_objects.go

```diff
@@ -107,7 +108,11 @@ type ConnectedObjectsQuery struct {
 	typesystem              *typesystem.TypeSystem
 	resolveNodeLimit        uint32
 	resolveNodeBreadthLimit uint32
-	maxResults              uint32
+
+	// candidateObjectsMap map allows us to not return the same object twice
```

Comment:
Slight change of verbiage here so it's more clear to the next developer

> candidateObjectsMap allows us to avoid expanding the same relationship more than once.

The bit "to not return the same object twice" implies avoiding sending duplicates, but it's not that we're avoiding sending duplicates, it's that we're avoiding expanding the same relationship more than once.

Reply:
No, `candidateObjectsMap` **is** avoiding sending duplicates

`visitedObjectsMap` is the one we use to avoiding expanding the same relationship more than once